src/compiler/checker.ts

```diff
@@ -8839,13 +8840,31 @@ namespace ts {
             })!.parent;
         }
 
+        function getInstanceTypeOfClassSymbol(classSymbol: Symbol): Type {
+            const classType = getDeclaredTypeOfSymbol(classSymbol) as InterfaceType;
+            if (!classType.typeParameters) {
+                return classType;
+            }
+            const typeArguments = map(classType.typeParameters, typeParameter => {
+                if (!inferInstanceTypeArgumentsAsConstraint) {
+                    return anyType;
+                }
+                const modifiers = getVarianceModifiers(typeParameter);

```

Comment:
Additionally, consider using `any` for `Independent` type parameters still (after all, it's unwitnessable and will probably save us work elsewhere), and for `Bivariant` parameters (since the wonky `in` _or_ `out` behavior that variance implies can't be modeled with strictly `unknown` or `never`). Ideally `Invariant` would a unique existential type, but, lacking that, hopefully `unknown` is accurate enough.

Reply:
Sounds good to me, will switch.